evalml/tests/automl_tests/test_dask_engine.py

```diff
@@ -0,0 +1,218 @@
+
+        # Verify all pipelines are trained and fitted.
+        seq_pipelines = fit_pipelines(pipelines, SequentialEngine())
+        for pipeline in seq_pipelines:
+            assert pipeline._is_fitted
+
+        # Verify all pipelines are trained and fitted.
+        par_pipelines = fit_pipelines(pipelines, DaskEngine(client=self.client))
+        for pipeline in par_pipelines:
+            assert pipeline._is_fitted
+
+        # Ensure sequential and parallel pipelines are equivalent
+        assert len(par_pipelines) == len(seq_pipelines)
+        for par_pipeline in par_pipelines:
+            assert par_pipeline in seq_pipelines
+
+    def test_submit_evaluate_job_single(self):
+        """ Test that evaluating a single pipeline using the parallel engine produces the
+        same results as simply running the evaluate_pipeline function. """
+        X, y = self.X_y_binary
+        X = ww.DataTable(X)
+        y = ww.DataColumn(y)
+        pipeline = TestLRCPipeline({})
+        engine = DaskEngine(client=self.client)
+
+        # Verify that engine evaluates a pipeline
+        pipeline_future = engine.submit_evaluation_job(X=X, y=y,
+                                                       automl_data=automl_data, pipeline=pipeline)
+        assert isinstance(pipeline_future, DaskComputation)
+
+        par_eval_results = pipeline_future.get_result()
+
+        original_eval_results = evaluate_pipeline(pipeline, automl_data=automl_data, X=X, y=y, logger=JobLogger())
+
+        # Ensure we get back the same output as the parallelized function.
+        assert len(par_eval_results) == 3
+
+        # Compare cross validation information except training time.
+        assert isinstance(par_eval_results[0], dict)
+        assert par_eval_results[0]["cv_data"] == original_eval_results[0]["cv_data"]
+        assert all(par_eval_results[0]["cv_scores"] == original_eval_results[0]["cv_scores"])
+        assert par_eval_results[0]["cv_score_mean"] == original_eval_results[0]["cv_score_mean"]
+
+        # Make sure the resulting pipelines are the same.
+        assert isinstance(par_eval_results[1], PipelineBase)
+        assert par_eval_results[1] == original_eval_results[1]
+
+        # Make sure a properly filled logger comes back.
+        assert isinstance(par_eval_results[2], JobLogger)
+        assert par_eval_results[2].logs == original_eval_results[2].logs
+
+    def test_submit_evaluate_jobs_multiple(self):
+        """ Test that evaluating multiple pipelines using the parallel engine produces the
+        same results as the sequential engine. """
+        X, y = self.X_y_binary
+        pipelines = [TestLRCPipeline({}),
+                     TestBaselinePipeline({}),
+                     TestSVMPipeline({})]
+
+        def eval_pipelines(pipelines, engine):
+            futures = []
+            for pipeline in pipelines:
+                futures.append(engine.submit_evaluation_job(X=ww.DataTable(X), y=ww.DataColumn(y),
```

Comment:
Yep our submit evaluation job assumes the data is woodwork objects. Since that assumption is not violated during AutoMLSearch and the accessor is coming soon, I think it's fine to not add coverage (and modify the code) to support dataframes!